### PR TITLE
#1552 Deployment Pull Backup

### PR DESCRIPTION
**Pull request by fantasya-pbem**

### CODE REVIEW

File: docs/deployment/pull-backup.rst [hunk 1]
```diff
@@ -11,6 +11,13 @@ the SSH connection must be established by the client via a remote tunnel. Other
 network file systems like NFS or SMB could be used as well, but SSHFS is very
 simple to set up and probably the most secure one.
 
+.. note::
+
+    There are user reports that SSHFS can be extremely slow over high latency
+    networks. Using SSHFS also means Borg does all the checksumming over the
+    network which is also very slow. You should benchmark your connection's
+    throughput first before deploying a huge setup.
```
Comment (ThomasWaldmann) [inline docs/deployment/pull-backup.rst]: hmm, this slightly misses the point.

the "checksumming" (chunking / hashing / lookup of the file contents) happens only if the files cache lookup considers the file as potentially changed (or new).

if that check works correctly (which might need fine tuning using --files-cache), it should work quite reliably. if a file is really changed, we need to read (chunk, ...) it anyway. if read speed in general is slow over sshfs, that is not a borg specific problem.

so, the reading (chunking, ...) is not the remarkable part that should be documented here, but that one needs to get the files cache working (what you already did below).

File: docs/deployment/pull-backup.rst [hunk 2]
```diff
@@ -68,10 +75,13 @@ create the backup, retaining the original paths, excluding the repository:
...
 
-    borg create --exclude /borgrepo /borgrepo::archive /
+    borg create --exclude /borgrepo --files-cache ctime,size /borgrepo::archive /
 
-(For the sake of simplicity only ``/borgrepo`` is excluded here. You may want to
-set up an exclude file with additional files and folders to be excluded.)
+For the sake of simplicity only ``/borgrepo`` is excluded here. You may want to
+set up an exclude file with additional files and folders to be excluded. Also
+note that we have to modify Borg's file change detection behaviour – SSHFS does
+not support real inode numbers, so we have to supply the ``--files-cache``
```
Comment (ThomasWaldmann) [inline docs/deployment/pull-backup.rst]: it doesn't matter whether they are "real", but rather whether they are stable.

so, they are not stable?

**Comment by ThomasWaldmann** [comment]
OK, thanks for writing this, guess this is merge ready?

Did you practically try this?

**Comment by fantasya-pbem** [comment]
Yes, I tested it on two VirtualBox VMs, everythings works as expected.

**Comment by ThomasWaldmann** [comment]
Merged! Thanks for your patience!

**Pull request merged**